src/types.jl

```diff
@@ -35,19 +35,20 @@ function get_registry(
 )
     if haskey(cache.registries, registry_url)
         registry_path = path(cache, registry_url)
+        default_branch = splitpath(String(readchomp(`git symbolic-ref refs/remotes/origin/HEAD`)))[end]
 
         if !ispath(registry_path)
             mkpath(path(cache))
-            run(`git clone $registry_url $registry_path --branch=master`)
+            run(`git clone $registry_url $registry_path --branch=$default_branch`)
         else
             # this is really annoying/impossible to do with LibGit2
             git = gitcmd(registry_path, gitconfig)
             run(`$git config remote.origin.url $registry_url`)
-            run(`$git checkout -q -f master`)
+            run(`$git checkout -q -f $default_branch`)
```

Comment:
In this side of the conditional I'm slightly less sure about the path forward. Perhaps we still clone the repo to a temp directory just to get the default branch? But that forces a clone even when there's no need which feels a bit annoying. What do you think @GunnarFarneback?

Reply:
Would it work to move the `git checkout` to the end of the `try` part of the `register` function and use `LibGit2.branch` to obtain the default branch in this function as well?